packages/@expo/cli/src/start/doctor/dependencies/getVersionedPackages.ts

```diff
@@ -123,17 +131,27 @@ export async function getVersionedPackagesAsync(
 
   let nativeModulesCount = 0;
   let othersCount = 0;
-  const excludedNativeModules: { name: string; bundledNativeVersion: string }[] = [];
+  const excludedNativeModules: ExcludedNativeModules[] = [];
 
   const versionedPackages = packages.map((arg) => {
-    const { name, type, raw } = npmPackageArg(arg);
+    const { name, type, raw, rawSpec } = npmPackageArg(arg);
 
     if (['tag', 'version', 'range'].includes(type) && name && versionsForSdk[name]) {
       // Unimodule packages from npm registry are modified to use the bundled version.
       // Some packages have the recommended version listed in https://exp.host/--/api/v2/versions.
-      if (pkg?.expo?.install?.exclude?.includes(name)) {
+      const isExcludedFromValidation = pkg?.expo?.install?.exclude?.includes(name);
+      const hasSpecifiedExactVersion = rawSpec !== '';
+      // React and React native should always be SDK compatible
+      const mustBeSDKCompaible = ['react', 'react-native'].includes(name);
```

Comment:
I made this explicit - we will support installing specific version of packages apart from React and React Native. They always get the SDK-compatible version (so same as now).

Reply:
i think we could remove this check. if people want to opt in to a potentially incompatible react/react-native version, they'll still be warned when they run `npx expo start` or `npx expo-doctor`